### PR TITLE
Lx 899 split og and indexing status

### PR DESCRIPTION
**Pull request by kzonov**
CS API around the `active` flag is far from being something reminding CRUD, so I made CUD actions to be at the end just update calls, and Read action is a dummy that is there only to be terraform compliant.
I'm going to ask CS maybe there is a way to fetch the status of indexing. If it is - I'll update the PR

### CODE REVIEW

File: chaossearch/resource_indexing_state.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	return &schema.Resource{
+		CreateContext: resourceIndexingStateCreate,
+		ReadContext:   resourceIndexingStateRead,
+		UpdateContext: resourceIndexingStateUpdate,
+		DeleteContext: resourceIndexingStateDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"objectGroupName": {
```
Comment (benzaita) [inline chaossearch/resource_indexing_state.go]: I think the convention is snake_case (at least in the other resource)

File: chaossearch/resource_indexing_state.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+	return resourceIndexingStateRead(ctx, data, meta)
+}
+
+func resourceIndexingStateRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+
+	// There is no request that is similar to getting an indexing state in ChaosSearch API,
+	//  so we cannot retrieve it, therefore this function is kept for the sake of being
+	//  compliant with Terraform providers standard, but it doesn't do anything
```
Comment (benzaita) [inline chaossearch/resource_indexing_state.go]: you can read it from the bucket tagging. I believe it's called "modeling" or something

Comment (kzonov) [reply chaossearch/resource_indexing_state.go]: https://docs.chaossearch.io/reference#get_v1-key doesn't look like 🤷 I asked CS whether it's possible

Comment (benzaita) [reply chaossearch/resource_indexing_state.go]: I think we already read it here: https://github.com/benzaita/terraform-provider-chaossearch/blob/master/chaossearch/client/operation_read_object_group.go#L108

Comment (kzonov) [reply chaossearch/resource_indexing_state.go]: seems that we're doing it wrong, as they use not a numerical representation there, but some words like "stopped"... I'm trying to figure it out with them

File: chaossearch/client/model.go [hunk 3]
```diff
@@ -53,3 +53,23 @@ type UpdateObjectGroupRequest struct {
...
+// There are other nested structures that we don't marshal here because we don't need to.
+//  Feel free to add them in the future when needed
+type BucketMetadata struct {
+	Bucket string `json:"Bucket"`
+	State  string `json:"State"`
+}
+
+type IndexingState struct {
+	ObjectGroupName string
+	Active          bool
```
Comment (benzaita) [inline chaossearch/client/model.go:68]: we should remove `Active` from `ReadObjectGroupResponse` now, right?

File: chaossearch/resource_indexing_state.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		ObjectGroupName: data.Get("object_group_name").(string),
+	}
+
+	resp, err := c.ReadIndexingState(ctx, readIndexingStateRequest)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.SetId(resp.ObjectGroupName)
+	data.Set("Active", resp.Active)
```
Comment (benzaita) [inline chaossearch/resource_indexing_state.go]:
```suggestion
	data.Set("active", resp.Active)
```

File: chaossearch/client/operation_update_indexing_state.go [hunk 5]
```diff
@@ -8,7 +8,8 @@ import (
 	"net/http"
 )
 
-func (client *Client) SetActive(ctx context.Context, req *SetActiveRequest) error {
+// For documentation see: https://docs.chaossearch.io/reference#bucketmodel
+func (client *Client) UpdateIndexingState(ctx context.Context, req *SetActiveRequest) error {
```
Comment (benzaita) [inline chaossearch/client/operation_update_indexing_state.go]: can we align the terms used in the method (indexing state) and struct (active)? I don't care in which way, but they both should be either "active" or "indexing state".

File: chaossearch/client/operation_read_indexing_state.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+// For documentation see: https://docs.chaossearch.io/reference#bucketmodel
+func (client *Client) ReadIndexingState(ctx context.Context, req *ReadIndexingStateRequest) (*IndexingState, error) {
+	method := "GET"
+
+	responseBody, err := makeGetBucketMetadataRequest(method, client, ctx)
+	if err != nil {
+		return nil, fmt.Errorf("request failed: %s", err)
+	}
+
+	var response ReadBucketMetadataResponse
```
Comment (benzaita) [inline chaossearch/client/operation_read_indexing_state.go]: Should the `ReadBucketMetadataResponse` struct really be public if the `makeGetBucketMetadataRequest` method is not?

File: chaossearch/client/operation_read_indexing_state.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+	if response.Metadata[req.ObjectGroupName].State == "Active" {
+		bucketMetadata.Active = true
+	}
+
+	return bucketMetadata, nil
+}
+
+func makeGetBucketMetadataRequest(method string, client *Client, ctx context.Context) (io.ReadCloser, error) {
+	url := fmt.Sprintf("%s/Bucket/metadata", client.config.URL)
```
Comment (benzaita) [inline chaossearch/client/operation_read_indexing_state.go:45]: did you find how is it different then [reading it from the bucket tagging](https://github.com/benzaita/terraform-provider-chaossearch/blob/master/chaossearch/client/operation_read_object_group.go#L108)?

**Review by benzaita — CHANGES_REQUESTED** [review]

File: chaossearch/client/model.go (hunk 3 above)
Comment (kzonov) [reply chaossearch/client/model.go:68]: Yeah. I was initially thinking to do it in a separate PR and a separate version bump, but it doesn't really make any sense to do it separately, so I'll remove it right now

File: chaossearch/client/operation_update_indexing_state.go (hunk 5 above)
Comment (kzonov) [reply chaossearch/client/operation_update_indexing_state.go]: True, I actually planned to rename it, simply forgot. Thanks for noticing, we definitely need to make it consistent

File: chaossearch/client/operation_read_indexing_state.go (hunk 7 above)
Comment (kzonov) [reply chaossearch/client/operation_read_indexing_state.go:45]: Not really, but it was what was suggested by Brian, so I thought it makes sense to use what he suggests

File: chaossearch/client/operation_read_indexing_state.go (hunk 6 above)
Comment (kzonov) [reply chaossearch/client/operation_read_indexing_state.go]: I was thinking that maybe in the future we'll extract this method to the client or smth like this. And this struct is not specific to this operation only, so it makes sense for it to be public. But I don't have a strong opinion on it and can change it if you want so

Comment (kzonov) [reply chaossearch/client/operation_read_indexing_state.go]: To not wait for a seaparate discussion I just did as you suggested :)

**Comment by kzonov** [comment]
Oh, I just realised that a call to get metadata expects a POST and not GET. Omg, it's all my biases and false expectations 🤦 :) I'll fix it

File: chaossearch/client/operation_read_indexing_state.go (hunk 6 above)
Comment (benzaita) [reply chaossearch/client/operation_read_indexing_state.go]: Either is fine with me. Thanks for addressing

**Review by benzaita — APPROVED** [review]

**Comment by benzaita** [comment]
great stuff!

**Pull request merged**